system/HTTP/CURLRequest.php

```diff
@@ -267,6 +265,68 @@ public function put(string $url, array $options = []): ResponseInterface
 
 	//--------------------------------------------------------------------
 
+	/**
+	 * Set the HTTP Authentication.
+	 *
+	 * @param string $username
+	 * @param string $password
+	 * @param string $type     basic or digest
+	 *
+	 * @return $this
+	 */
+	public function setAuth(string $username, string $password, string $type = 'basic')
+	{
+		$this->config['auth'] = [
+			$username,
+			$password,
+			$type,
+		];
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set form data to be sent.
+	 *
+	 * @param array   $params
+	 * @param boolean $multipart Set TRUE if you are sending CURLFiles
+	 *
+	 * @return $this
+	 */
+	public function setForm(array $params, bool $multipart = false)
+	{
+		if ($multipart)
+		{
+			$this->config['multipart'] = $params;
+		}
+		else
+		{
+			$this->config['form_params'] = $params;
+		}
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set JSON data to be sent.
+	 *
+	 * @param mixed $data
+	 *
+	 * @return $this
+	 */
+	public function setJSON($data)
+	{
+		$this->config['json'] = $data;
```

Comment:
If we're setting json in the request, shouldn't we also set `Content-Type: application/json`?

Reply:
We could update this to use the `\Config\Format::getFormatter('application/json')`.

Allow to use a custom JSON Formatter...

Reply:
I don't think there is any intent to format the JSON data. If anything, it might be stringify'd. The unit testing should make this clearer.